crates/configuration/src/network.rs

```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
 
     pub fn dump_to_toml(&self) -> Result<String, toml::ser::Error> {
         // This regex is used to replace the "" enclosed u128 value to a raw u128 because u128 is not supported for TOML serialization/deserialization.
-        let re = Regex::new(r#""U128%(?<u128_value>\d+)""#).expect("regex should be valid, this is a bug please report it: https://github.com/paritytech/zombienet-sdk/issues");
+        let re = Regex::new(r#""U128%(?<u128_value>\d+)""#)
+            .expect(&format!("{} {}", VALID_REGEX, THIS_IS_A_BUG));
         let toml_string = toml::to_string_pretty(&self)?;
 
         Ok(re.replace_all(&toml_string, "$u128_value").to_string())
     }
+
+    pub fn load_from_toml(path: &str) -> Result<NetworkConfig, anyhow::Error> {
+        let file_str = fs::read_to_string(path).expect(&format!("{} {}", RW_FAILED, THIS_IS_A_BUG));
+        let re: Regex = Regex::new(r"(?<field_name>(initial_)?balance)\s+=\s+(?<u128_value>\d+)")
+            .expect(&format!("{} {}", VALID_REGEX, THIS_IS_A_BUG));
+
+        let mut network_config: NetworkConfig = toml::from_str(
+            re.replace_all(&file_str, "$field_name = \"$u128_value\"")
+                .as_ref(),
+        )?;
+
+        if network_config.relaychain.is_none() {
+            Err(anyhow!("Relay chain does not exist."))?
+        }
+
+        // retrieve the defaults relaychain for assigning to nodes if needed
+        let relaychain_default_command: Option<Command> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_command()
+            .cloned();
+
+        let relaychain_default_image: Option<Image> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_image()
+            .cloned();
+
+        let relaychain_default_db_snapshot: Option<crate::shared::types::AssetLocation> =
+            network_config
+                .relaychain
+                .as_ref()
+                .unwrap()
+                .default_db_snapshot()
+                .cloned();
+
+        let default_args: Vec<Arg> = network_config
```

Comment:
We are using two ways here to access the data
`.relaychain.as_ref().unwrap()` and `.relaychain()` we should access in a uniform way here.

Reply:
I convert all to `.relaychain()` as it makes more sense than using `unwrap()`